### PR TITLE
Replicate RemoteUnitTestExecutor into repository.

### PR DESCRIPTION
**Pull request by jander-msft**
This change replicates the subset of the RemoteUnitTestExecutor package that is used by the Instrumentation Engine tests to avoid usage of delay signed assemblies and missing dependencies. I've also fixed the TestGetConfigurationPaths test to unset its environment variables that are set during the test in order to avoid impact on tests that follow it in execution order.

### CODE REVIEW

File: src/Tests/RemoteUnitTestExecutor.Host_x86/RemoteUnitTestExecutor.Host_x86.csproj [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (jander-msft) [inline src/Tests/RemoteUnitTestExecutor.Host_x86/RemoteUnitTestExecutor.Host_x86.csproj:1]: I will add another host project for x64 to avoid loading x64 assembly into 32-bit processes.

File: InstrumentationEngine.sln [hunk 2]
```diff
@@ -588,6 +592,54 @@ Global
...
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Managed|x64.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Managed|x64.Build.0 = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Managed|x86.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Managed|x86.Build.0 = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|Any CPU.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|Any CPU.Build.0 = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|x64.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|x64.Build.0 = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|x86.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|x86.Build.0 = Debug|Any CPU
```
Comment (delmyers) [inline InstrumentationEngine.sln:630]: Don't build managed projects as part of the native code analysis

File: InstrumentationEngine.sln [hunk 3]
```diff
@@ -588,6 +592,54 @@ Global
...
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Managed|x64.ActiveCfg = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Managed|x64.Build.0 = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Managed|x86.ActiveCfg = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Managed|x86.Build.0 = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Native|Any CPU.ActiveCfg = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Native|Any CPU.Build.0 = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Native|x64.ActiveCfg = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Native|x64.Build.0 = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Native|x86.ActiveCfg = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Native|x86.Build.0 = Debug|Any CPU
```
Comment (delmyers) [inline InstrumentationEngine.sln:606]: Don't build managed projects as part of the native code analysis.

File: InstrumentationEngine.sln [hunk 4]
```diff
@@ -588,6 +592,54 @@ Global
 		{9F97B063-5ACF-409A-8DCC-4BD9FF817D1D}.Release|x64.Build.0 = Release|Any CPU
 		{9F97B063-5ACF-409A-8DCC-4BD9FF817D1D}.Release|x86.ActiveCfg = Release|Any CPU
 		{9F97B063-5ACF-409A-8DCC-4BD9FF817D1D}.Release|x86.Build.0 = Release|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Managed|Any CPU.ActiveCfg = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Managed|Any CPU.Build.0 = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Managed|x64.ActiveCfg = Debug|Any CPU
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Managed|x64.Build.0 = Debug|Any CPU
```
Comment (delmyers) [inline InstrumentationEngine.sln:598]: Only build managed code analysis as in "Any CPU" builds.

File: InstrumentationEngine.sln [hunk 5]
```diff
@@ -588,6 +592,54 @@ Global
...
+		{182D00C1-705F-4FF6-8620-8EEF29936C7A}.CodeAnalysis.Managed|x86.ActiveCfg = Debug|Any CPU
```
Comment (delmyers) [inline InstrumentationEngine.sln:599]: >Debug [](http://example.com/codeflow?start=78&length=5)

Build Code Analysis as "Release"

File: InstrumentationEngine.sln [hunk 6]
```diff
@@ -588,6 +592,54 @@ Global
...
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|x64.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|x64.Build.0 = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|x86.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.CodeAnalysis.Native|x86.Build.0 = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.Debug|x64.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.Debug|x64.Build.0 = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.Debug|x86.ActiveCfg = Debug|Any CPU
+		{F356AF58-F432-415D-BE13-44228F265CEF}.Debug|x86.Build.0 = Debug|Any CPU
```
Comment (delmyers) [inline InstrumentationEngine.sln:636]: Do these need to be built as part of the platform specific builds?

File: src/Tests/RemoteUnitTestExecutor/AssertExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+using System.Collections.Generic;
+
+namespace RemoteUnitTestExecutor
+{
+    public static class AssertExtensions
+    {
+        public static void ClearInvokedMethods(IList<MethodInvocationInfo> invokedMethods)
+        {
+            invokedMethods.Clear();
+            Assert.AreEqual<int>(0, invokedMethods.Count, "Invoked methods are not cleared.");
```
Comment (delmyers) [inline src/Tests/RemoteUnitTestExecutor/AssertExtensions.cs:14]: >0, invokedMethods.Count [](http://example.com/codeflow?start=33&length=23)

How could it be possible for this to be not true? Wouldn't it indicate some kind of race condition that should be accounted for elsewhere?

File: src/Tests/RemoteUnitTestExecutor/Program.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace RemoteUnitTestExecutor
+{
+    public class Program
+    {
+        [LoaderOptimization(LoaderOptimization.MultiDomainHost)]
```
Comment (delmyers) [inline src/Tests/RemoteUnitTestExecutor/Program.cs:10]: >[LoaderOptimization(LoaderOptimization.MultiDomainHost)] [](http://example.com/codeflow?start=8&length=56)

This is interesting, and only meaningful in .NET framework. Why is this here?

File: InstrumentationEngine.sln (hunk 6 above)
Comment (jander-msft) [reply InstrumentationEngine.sln:636]: The other projects that rely on the RemoteUnitTestExecutor already build for platform specific builds, so I wanted to keep these the same. In the hopefully near future, I'd like to make all of the managed projects be platform agnostic, which will simplify restoring and building.

File: src/Tests/RemoteUnitTestExecutor/Program.cs (hunk 8 above)
Comment (jander-msft) [reply src/Tests/RemoteUnitTestExecutor/Program.cs:10]: The RemoteUnitTestExecutor is only for .NET Framework tests. This assembly and the test assemblies that depend on it build for .NET Framework 4.5. Why this is here, I'm not certain, but its in the original code.

File: src/Tests/RemoteUnitTestExecutor/AssertExtensions.cs (hunk 7 above)
Comment (jander-msft) [reply src/Tests/RemoteUnitTestExecutor/AssertExtensions.cs:14]: This is what is written in the original code and I didn't want to change it unless I have to.

**Review by WilliamXieMSFT — APPROVED** [review]
:shipit:
